sw/device/lib/dif/dif_pinmux.c

```diff
@@ -5,19 +5,187 @@
+      break;
+    case kDifPinmuxLockTargetOutsel:
+      reg_base = PINMUX_MIO_OUTSEL_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_MIO_OUTSEL_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetMioSleep:
+      reg_base = PINMUX_MIO_PAD_SLEEP_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_MIO_PAD_SLEEP_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetDioSleep:
+      reg_base = PINMUX_DIO_PAD_SLEEP_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_DIO_PAD_SLEEP_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetMioPadAttr:
+      reg_base = PINMUX_MIO_PAD_ATTR_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_MIO_PAD_ATTR_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetDioPadAttr:
+      reg_base = PINMUX_DIO_PAD_ATTR_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_DIO_PAD_ATTR_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetWakeupDetector:
+      reg_base = PINMUX_WKUP_DETECTOR_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_WKUP_DETECTOR_REGWEN_MULTIREG_COUNT;
+      break;
+    default:
+      return kDifBadArg;
+  }
+
+  if (index >= reg_count) {
+    return kDifBadArg;
+  }
+
+  uintptr_t reg_offset = reg_base + index * sizeof(uint32_t);
+  mmio_region_write32(pinmux->base_addr, reg_offset, /*value=*/0);
+  return kDifOk;
+}
+
+dif_result_t dif_pinmux_is_locked(const dif_pinmux_t *pinmux,
+                                  dif_pinmux_index_t index,
+                                  dif_pinmux_lock_target_t target,
+                                  bool *is_locked) {
+  if (pinmux == NULL || is_locked == NULL) {
+    return kDifBadArg;
+  }
+
+  uintptr_t reg_base;
+  uint32_t reg_count;
+  switch (target) {
+    case kDifPinmuxLockTargetInsel:
+      reg_base = PINMUX_MIO_PERIPH_INSEL_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_MIO_PERIPH_INSEL_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetOutsel:
+      reg_base = PINMUX_MIO_OUTSEL_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_MIO_OUTSEL_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetMioSleep:
+      reg_base = PINMUX_MIO_PAD_SLEEP_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_MIO_PAD_SLEEP_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetDioSleep:
+      reg_base = PINMUX_DIO_PAD_SLEEP_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_DIO_PAD_SLEEP_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetMioPadAttr:
+      reg_base = PINMUX_MIO_PAD_ATTR_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_MIO_PAD_ATTR_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetDioPadAttr:
+      reg_base = PINMUX_DIO_PAD_ATTR_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_DIO_PAD_ATTR_REGWEN_MULTIREG_COUNT;
+      break;
+    case kDifPinmuxLockTargetWakeupDetector:
+      reg_base = PINMUX_WKUP_DETECTOR_REGWEN_0_REG_OFFSET;
+      reg_count = PINMUX_WKUP_DETECTOR_REGWEN_MULTIREG_COUNT;
```

Comment:
maybe it would be better to move this to a function or X-macro setup, since you repeat the same code above (L76-L111)?